Add missing occ encryption commands (#3727)
owncloud/docs: modules/admin_manual/pages/configuration/server/occ_commands/core_commands/encryption_commands.adoc

= Encryption

> `occ` includes a complete set of commands for managing encryption. When using a HSM (Hardware Security Module, can also be emulated by software), additional occ encryption related commands can be used.

Reviewer: ```suggestion
`occ` includes a complete set of commands for managing encryption. When using a HSM (Hardware Security Module, can also be emulated by software), additional occ encryption-related commands can be used.
```

 encryption:status                   Lists the current status of encryption
----

> Commands only available when using HSM. For details see the occ HSM occ documentation below.

Reviewer: ```suggestion
Commands only available when using HSM. For details see the HSM occ documentation below.
```

The `-y` switch can be supplied to automate acceptance of user input.

== HSM Related Commands

=== Export or Import the Masterkey

[source,console,subs="attributes+"]
----
{occ-command-example-prefix} encryption:hsmdaemon [options]
----

=== Options

[width="100%",cols="20%,70%",]
|===
| `--export-masterkey` | Export the private master key in base64
| `--import-masterkey= +
IMPORT-MASTERKEY`     | Import a base64 encoded private masterkey.
|===

=== Decrypt a String

[source,console,subs="attributes+"]
----
{occ-command-example-prefix} encryption:hsmdaemon:decrypt [options] [--] <decrypt>
----

=== Arguments

[width="100%",cols="20%,70%",]
|===
| `decrypt` | The string to decrypt
|===

=== Options

[width="100%",cols="20%,70%",]
|===
| `--username[=USERNAME]` | The name of the user who is able to decrypt the provided string
| `--keyId[=KEYID]`       | The keyId which was used to encrypt the provided string
|===

=== Set the HSM URL

Set the url according your configuration

[source,console,subs="attributes+"]
----
{occ-command-example-prefix} config:app:set encryption hsm.url --value 'http://localhost:8513'
----

=== Set the JSON Web Token Secret

> Set the secret according your configuration

Reviewer: ```suggestion
Set the secret according to your configuration
```

[source,console,subs="attributes+"]
----
{occ-command-example-prefix} config:app:set encryption hsm.jwt.secret --value '7a7d1826-b514-4d9f-afc7-a7485084e8de'
----

=== Set the JWT Clockskew

> Set or change the clockskew only if ownCloud advices to do so. Value is in seconds

Reviewer 2: ```suggestion
Set or change the clockskew only if ownCloud advises to do so. Value is in seconds
```

> Set or change the clockskew only if ownCloud advices to do so. Value is in seconds

Reviewer: ```suggestion
Set or change the clockskew only if ownCloud advises you to do so. Value is in seconds
```
